File: PYTHON/Program/Keluarga.py
from Ayah import Ayah
from Ibu import Ibu
from Anak import Anak

class Keluarga:
    def __init__(self, nama_keluarga="", alamat_rumah=""):
        self.nama_keluarga = nama_keluarga
        self.alamat_rumah = alamat_rumah
        self.ayah = None
        self.ibu  = None
        self.anak = []

    #getter dan setter
    def getNamaKeluarga(self):
        return self.nama_keluarga

    def setNamaKeluarga(self, nama_keluarga):
        self.nama_keluarga = nama_keluarga

    def getAlamatRumah(self):
        return self.alamat_rumah

    def setAlamatRumah(self, alamat_rumah):
        self.alamat_rumah = alamat_rumah
    
    #untuk menambahkan data ayah, ibu, dan anak
    def setAyah(self, a: Ayah):
        self.ayah = a

    def setIbu(self, i: Ibu):
        self.ibu = i

    def add_anak(self, an: Anak):
        self.anak.append(an)

    #untuk menampilkan data 
    def print(self):
        print("=== Data Keluarga ===")
        print(f"Nama Keluarga : {self.nama_keluarga}")
        print(f"Alamat Rumah  : {self.alamat_rumah}")

        if self.ayah:
                print("\n=== Data Ayah ===")
                self.ayah.print_ayah()
        else:
            print("===Belum ada data ayah===\n")
        if self.ibu:
           
                print("\n=== Data Ibu ===")
                self.ibu.print_ibu()
        else:
            print("===Belum ada data ibu===\n")

      
        if self.anak:
            for an in self.anak:
                print("\n=== Data Anak ===")
                an.print_anak()
        else:
            print("===Belum ada data anak==")

File: JAVA/Program/Keluarga.java
import java.util.ArrayList;
public class Keluarga{
    private String namaKeluarga;
    private String alamatRumah;
    private Ayah ayah;
    private Ibu ibu;
    private ArrayList<Anak> anak;

    //untuk pengecekan ada ayah ibu
    private boolean adaAyah = false;  
    private boolean adaIbu = false;
    
    //constructor
    public Keluarga(){
    }   

    public Keluarga(String namaKeluarga, String alamatRumah) {
        this.namaKeluarga = namaKeluarga;
        this.alamatRumah = alamatRumah;
        this.anak = new ArrayList<>();
    }

    //getter dan setter
     public String getNamaKeluarga() {
        return namaKeluarga;
    }

    public void setNamaKeluarga(String namaKeluarga) {
        this.namaKeluarga = namaKeluarga;
    }

    public String getAlamatRumah() {
        return alamatRumah;
    }

    public void setAlamatRumah(String alamatRumah) {
        this.alamatRumah = alamatRumah;
    }

    public void setAyah(Ayah a){
        this.ayah = a;
        this.adaAyah = true;
    }

    public void setIbu(Ibu i){
        this.ibu = i;
        this.adaIbu = true;
    }

    //untuk menambahkan anak
    public void addAnak(Anak an) {
        this.anak.add(an);
    }

    //untuk menampilkan data 
      public void print() {
        System.out.println("=== Data Keluarga ===");
        System.out.println("Nama Keluarga : " + namaKeluarga);
        System.out.println("Alamat Rumah  : " + alamatRumah);

        if (adaAyah) {
            System.out.println("=== Data Ayah ===");
            ayah.printAyah();
        } else {
            System.out.println("Data Ayah Kosong\n");
        }

        if (adaIbu) {
            System.out.println("=== Data Ibu ===");
            ibu.printIbu();
        } else {
            System.out.println("Data Ibu Kosong\n");
        }

        if (!anak.isEmpty()) {
            for (Anak an : anak) {
                System.out.println("=== Data Anak ===");
                an.printAnak();
            }
        } else {
            System.out.println("Data Anak Kosong\n");
        }
    }
}